scipy/special/_support_cupy_torch_jax.py

```diff
@@ -0,0 +1,50 @@
+import sys
+from scipy._lib.array_api_compat.array_api_compat import array_namespace
+from . import _ufuncs
+from ._ufuncs import *
+
+__all__ = _ufuncs.__all__
+
+array_api_comat_prefix = "scipy._lib.array_api_compat.array_api_compat"
+
+
+def get_array_special_func(f_name, xp):
+    if xp.__name__ == f"{array_api_comat_prefix}.numpy":
+        f = getattr(_ufuncs, f_name, None)
+    elif xp.__name__ == f"{array_api_comat_prefix}.torch":
+        f = getattr(xp.special, f_name, None)
+    elif xp.__name__ == f"{array_api_comat_prefix}.cupy":
+        import cupyx
+        f = getattr(cupyx.scipy.special, f_name, None)
+    elif xp.__name__ == f"{array_api_comat_prefix}.jax":
+        f = getattr(xp.scipy.special, f_name, None)
+    else:
+        f = None
+
+    if f is None:
+        raise ValueError("SciPy cannot dispatch to special function "
+                         f"'{f_name}' in array library '{xp.__name__}'.")
+    return f
+
+
+# functools.wraps doesn't work because:
+# 'numpy.ufunc' object has no attribute '__module__'
+def support_cupy_torch_jax(f_name, n_args):
+    def wrapped(*args, **kwargs):
+        xp = array_namespace(*args[:n_args])
+        f = get_array_special_func(f_name, xp)
+        return f(*args, **kwargs)
+    wrapped.__doc__ = getattr(_ufuncs, f_name).__doc__
```

Comment:
Probably best not to do that in this decorator. Programmatically appending notes is what we tried in various flavors before, and it usually doesn't work well without a ton of effort. For now we can keep it in the module-level docstring for `special` I'd say.

Reply:
@rgommers let me know how you were thinking this documentation should go and I'll add it here. Or if you're open to seeing how docscrape would do, LMK what you'd like to see in the docs (notes section, or more prominently?) of each wrapped function. 

Maybe something like 

> This function has preliminary support for CuPy, PyTorch, and JAX arrays. When environment variable `SCIPY_ARRAY_API=1`, arrays of these types are automatically passed to the appropriate function of the native library for calculation, and it will return an array of the same type. In this case, only the first positional argument it accepted.

Reply:
That looks like a nice doc snippet to me. The last two sentences may perhaps be confusing to the average reader. In most cases there aren't more than 1 (or X) arguments, right? So everything is actually tested in those cases, and those sentences may be left out?